WcaOnRails/app/controllers/server_status_controller.rb

```diff
@@ -2,6 +2,22 @@
 class ServerStatusController < ApplicationController
   MINUTES_IN_WHICH_A_JOB_SHOULD_HAVE_STARTED_RUNNING = 5
 
+  def self.bad_keys_by_type_by_locale
+    @bad_keys_by_type_by_locale ||= begin
+      bad_keys_by_type_by_locale = {}
```

Comment:
I like that, although I'm not I'm not in love with the variable named `hash`. I also don't love that `bad_keys` sounds like an array, but I think that's my own problem I need to get over (sometimes I fear that my variable naming preferences have me on a path that ends in Hungarian notation).

Still, this solves a real problem, and I think it's better than my implementation, so I'm going to merge it.